Documentation for the Broker support access (#4327)
EnMasseProject/enmasse: documentation/modules/proc-troubleshooting.adoc

{cmdcli} logs deployment/admin._uuid_ -c standard-controller
{cmdcli} logs deployment/admin._uuid_ -c agent
----

= Examining the state of a broker using the {BrokerName} management interfaces

> If a problem is suspected with a Broker associated with an address space, it is possible to examine the state of the

Reviewer: ```suggestion
If a problem is suspected with a Broker associated with an address space, you can examine the state of the
```

> broker directly using its in-built link:{BrokerManagementDocs}[management interfaces].  {ProductName} exposes the

Reviewer: ```suggestion
broker directly using its built-in link:{BrokerManagementDocs}[management interfaces].  {ProductName} exposes the
```

{BrokerName}'s CLI and JMX  (via Jolokia).  It does not expose the {BrokerName} Console.

.Procedure

ifeval::["{cmdcli}" == "oc"]
. Log in as a service admin:
+
[subs="attributes",options="nowrap"]
----
{cmdcli} login -u admin
----

. Change to the project where {ProductName} is installed:
+
[subs="+quotes,attributes",options="nowrap"]
----
{cmdcli} project _{ProductNamespace}_
----
endif::[]

. Retrieve the uuid for the address space:
+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get addressspace _myspace_ -o jsonpath='{.metadata.annotations.enmasse\.io/infra-uuid}'
----

. Retrieve the broker support credentials (username and password) for the address space
+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get secret broker-support-_uuid_  --template='{{.data.username}}' | base64 --decode
{cmdcli} get secret broker-support-_uuid_  --template='{{.data.password}}' | base64 --decode
----

> . Identity the broker pod name

Reviewer: ```suggestion
. Identify the broker pod name:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get pods -l infraUuid=_uuid_,role=broker
----
+

> In the standard address there may be many brokers.  To identify the broker(s) hosting a particular queue, use this command:

Reviewer: ```suggestion
In the standard address, there may be many brokers.  To identify the broker(s) hosting a particular queue, use this command:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} get address _address resource name_  -o jsonpath="{.status.brokerStatuses[*].containerId}"
----
+

> . Execute support commands on the broker's pod

Reviewer: ```suggestion
. Execute support commands on the broker's pod:
```

+

> To execute an {BrokerName} Jolokia JMX commands, use a command line like this:

Reviewer: ```suggestion
To execute an {BrokerName} Jolokia JMX command, use a command similar to the following:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} exec _broker pod name_ -- {BrokerCLI} address show --user _username_ --password _password_
----
+

> To execute an {BrokerName} CLI commands, use a command line like this:

Reviewer: ```suggestion
To execute an {BrokerName} CLI command, use a command similar to the following:
```

+
[options="nowrap",subs="+quotes,attributes"]
----
{cmdcli} oc exec _broker pod name_ -- curl --silent --insecure --user _username_:_password_ 'https://localhost:8161/console/jolokia/read/org.apache.activemq.artemis:broker="_broker pod name_"/AddressMemoryUsage'
----
+
IMPORTANT: The double quotes around the broker pod name within the URL are required. Make sure you protect them from your

> command shell (as done above using the single quotes surrounding the whole URL). If they are not present, you will

Reviewer: ```suggestion
command shell using single quotes surrounding the whole URL, as shown in the above command. If they are not present, you will
```
